x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/registry/ClearInferenceEndpointCacheAction.java

```diff
@@ -0,0 +1,239 @@
+ */
+public class ClearInferenceEndpointCacheAction extends AcknowledgedTransportMasterNodeAction<ClearInferenceEndpointCacheAction.Request> {
+    private static final Logger log = LogManager.getLogger(ClearInferenceEndpointCacheAction.class);
+    private static final String NAME = "cluster:admin/xpack/inference/clear_inference_endpoint_cache";
+    public static final ActionType<AcknowledgedResponse> INSTANCE = new ActionType<>(NAME);
+    private static final String TASK_QUEUE_NAME = "inference-endpoint-cache-management";
+
+    private final ProjectResolver projectResolver;
+    private final InferenceEndpointRegistry inferenceEndpointRegistry;
+    private final MasterServiceTaskQueue<RefreshCacheMetadataVersionTask> taskQueue;
+
+    @Inject
+    public ClearInferenceEndpointCacheAction(
+        TransportService transportService,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        ProjectResolver projectResolver,
+        InferenceEndpointRegistry inferenceEndpointRegistry
+    ) {
+        super(
+            NAME,
+            transportService,
+            clusterService,
+            threadPool,
+            actionFilters,
+            ClearInferenceEndpointCacheAction.Request::new,
+            EsExecutors.DIRECT_EXECUTOR_SERVICE
+        );
+        this.projectResolver = projectResolver;
+        this.inferenceEndpointRegistry = inferenceEndpointRegistry;
+        this.taskQueue = clusterService.createTaskQueue(TASK_QUEUE_NAME, Priority.IMMEDIATE, new CacheMetadataUpdateTaskExecutor());
+        clusterService.addListener(
+            event -> event.state()
+                .metadata()
+                .projects()
+                .values()
+                .stream()
+                .map(ProjectMetadata::id)
+                .filter(id -> event.customMetadataChanged(id, InvalidateCacheMetadata.NAME))
+                .peek(id -> log.trace("Trained model cache invalidated on node [{}]", () -> event.state().nodes().getLocalNodeId()))
+                .forEach(inferenceEndpointRegistry::invalidateAll)
+        );
+    }
+
+    @Override
+    protected void masterOperation(
+        Task task,
+        ClearInferenceEndpointCacheAction.Request request,
+        ClusterState state,
+        ActionListener<AcknowledgedResponse> listener
+    ) {
+        if (inferenceEndpointRegistry.cacheEnabled()) {
+            taskQueue.submitTask("invalidateAll", new RefreshCacheMetadataVersionTask(projectResolver.getProjectId(), listener), null);
+        } else {
+            listener.onResponse(AcknowledgedResponse.TRUE);
+        }
+    }
+
+    @Override
+    protected ClusterBlockException checkBlock(ClearInferenceEndpointCacheAction.Request request, ClusterState state) {
+        return state.blocks().globalBlockedException(ClusterBlockLevel.METADATA_WRITE);
```

Comment:
Any idea when we should use the different `ClusterBlockLevel` enums? Are we using metadata here vs `WRITE` because we don't actually modify an index?